apps/api/prisma/schema.prisma

```diff
@@ -252,3 +252,269 @@ model RawTreasureAccount {
   @@ignore
   @@schema("smart_account")
 }
+
+// Gems.
+model GemsSummary {
+  id String @id @default(cuid())
+
+  userId        String? @unique @map("user_id")
+  legacyAddress String? @unique @map("legacy_address") @db.VarChar(42)
+
+  gems         Int @default(0)
+  gemsEarned   Int @default(0) @map("gems_earned")
+  shards       Int @default(0)
+  shardsEarned Int @default(0) @map("shards_earned")
+
+  createdAt DateTime @default(now()) @map("created_at") @db.Timestamptz(6)
+  updatedAt DateTime @map("updated_at") @db.Timestamptz(6)
+
+  @@index([gems])
+  @@index([gemsEarned])
+  @@index([shards])
+  @@index([shardsEarned])
+  @@index([createdAt])
+  @@index([updatedAt])
+  @@map("gems_summary")
+  @@schema("public")
+}
+
+model GemsTx {
+  txId String @id @map("tx_id")
+
+  userId        String? @map("user_id")
+  legacyAddress String? @map("legacy_address")
+
+  category     String
+  amount       Int     @default(0)
+  amountShards Int     @default(0) @map("amount_shards")
+  description  String
+  internalNote String? @map("internal_note")
+
+  createdAt DateTime @default(now()) @map("created_at") @db.Timestamptz(6)
+
+  @@index([userId])
+  @@index([legacyAddress])
+  @@index([category])
+  @@index([amount])
+  @@index([amountShards])
+  @@index([createdAt])
+  @@map("gems_tx")
+  @@schema("public")
+}
+
+// Rewards/Chests/Vouchers.
+enum RewardType {
+  Nft
+  Coin
+
+  @@map("reward_type_enum")
+  @@schema("public")
+}
+
+model Reward {
+  id           String     @id
+  type         RewardType
+  subType      String?    @map("sub_type")
+  chain        String
+  tokenAddress String     @map("token_address")
+  createdAt    DateTime   @default(now()) @map("created_at") @db.Timestamptz(6)
+  updatedAt    DateTime   @default(now()) @updatedAt @map("updated_at")
+
+  // NFT Type only
+  slug     String?
+  tokenId  String? @map("token_id")
+  quantity Int?
+
+  // Coin Type only
+  quantityWei String? @map("quantity_wei")
+
+  // Relations.
+  chestTypeRewards ChestTypeReward[]
+
+  @@map("rewards")
```

Comment:
Bit of inconsistency with the singular vs plural table names here - let's just keep them all singular